openshift-kube-apiserver/filters/deprecatedapirequest/v1helpers/helpers.go

```diff
@@ -14,43 +13,58 @@ import (
 
 type UpdateStatusFunc func(maxNumUsers int, status *apiv1.APIRequestCountStatus)
 
-func ApplyStatus(ctx context.Context, client apiclientv1.APIRequestCountInterface, name string, updateFuncs ...UpdateStatusFunc) (*apiv1.APIRequestCountStatus, bool, error) {
+func ApplyStatus(ctx context.Context, client apiv1client.APIRequestCountInterface, name string, statusDefaulter UpdateStatusFunc, updateFuncs ...UpdateStatusFunc) (*apiv1.APIRequestCountStatus, bool, error) {
 	updated := false
 	var updatedStatus *apiv1.APIRequestCountStatus
 	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		check, err := client.Get(ctx, name, metav1.GetOptions{})
+		apiRequestCount, err := client.Get(ctx, name, metav1.GetOptions{})
 		if errors.IsNotFound(err) {
-			// on a not found, let's create this thing.
-			requestCount := &apiv1.APIRequestCount{
+			// APIRequestCount might have been purposely deleted. We will
+			// try to create it again further below if there is a need to.
+			apiRequestCount = &apiv1.APIRequestCount{
 				ObjectMeta: metav1.ObjectMeta{Name: name},
-				Spec: apiv1.APIRequestCountSpec{
-					NumberOfUsersToReport: 10,
-				},
+				Spec:       apiv1.APIRequestCountSpec{NumberOfUsersToReport: 10},
 			}
-			check, err = client.Create(ctx, requestCount, metav1.CreateOptions{})
-		}
-		if err != nil {
+			// make sure the status doesn't result in a diff on a no-op.
+			statusDefaulter(10, &apiRequestCount.Status)
+		} else if err != nil {
 			return err
 		}
-		oldStatus := check.Status
+		oldStatus := apiRequestCount.Status
 		newStatus := oldStatus.DeepCopy()
 		for _, update := range updateFuncs {
-			update(int(check.Spec.NumberOfUsersToReport), newStatus)
+			update(int(apiRequestCount.Spec.NumberOfUsersToReport), newStatus)
```

Comment:
doesn't this `apiRequestCount` that you're updating here for new values end up overwritten below?  Is the re-use ok because we stored the output in `newStatus`?

Can we just rename the one below so that readers like me dont' get confused?

Reply:
Done.